clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java

```diff
@@ -37,97 +31,57 @@
 import tachyon.thrift.InvalidPathException;
 
 /**
- * Tachyon File System client. This class is the entry point for all file level operations on
- * Tachyon files. An instance of this class can be obtained via {@link TachyonFileSystem#get}. This
- * class is thread safe. The read/write interface provided by this client is similar to Java's
- * input/output streams.
+ * A TachyonFileSystem implementation including convenience methods as well as a streaming API to
+ * read and write files. Creating files is not supported by this TachyonFileSystem because the
+ * files should only be written once, thus getOutStream is sufficient for creating and writing to
+ * a file.
  */
 @PublicApi
-public final class TachyonFileSystem implements Closeable, TachyonFSCore {
-  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+public class TachyonFileSystem extends AbstractTachyonFileSystem {
```

Comment:
Is this class NOT supposed to call the master, and all the master calls should happen in the AbstractTachyonFileSystem?